### ISSUE TITLE
Android client: accept share of multiple URLs

### ISSUE THREAD

**Issue by zackw** [wallabag/android-app#1419]
**Is your feature request related to a problem? Please describe.**

I'm using wallabag Android client v2.5.3 (233).

I have a giant pile of tabs open on my phone that I would like to move into wallabag.  As far as I can tell, the only way to do this is one at a time: select each tab, tap the menu, tap share, tap the wallabag icon, and then tap the little square with the arrow on the pop-up for editing the title and tags.  Since I have nearly 100 tabs open, this is incredibly tedious.

**Describe the solution you'd like**

I would like this to work: go into the tab overview screen, select all the tabs I want to move to wallabag, share -> wallabag, *done*.  In this mode it should not bother prompting to edit the title or the tags, it should just send all the URLs straight to the server.  Assume the user will sort out the labeling later.

If you do this now, what happens is it brings up the pop-up for the first tab and ignores the rest of the URLs, which is more than a little surprising.

**Comment by di72nn** [comment]
> and then tap the little square with the arrow on the pop-up for editing the title and tags.

This is completely optional, you can just dismiss that pop-up immediately.

At this point the Android app doesn't support bulk-import, and I'm not sure whether any proper way of bulk-importing exists for the wallabag server, but if it's a one-time procedure, I would suggest you writing a simple script to iterate over your URLs and sending them to the wallabag server through the API. There are a few [API clients for that](https://doc.wallabag.org/en/developer/api/resources) (for your case [this Go implementation](https://codeberg.org/strubbl/wallabag-add-article) or [my python implementation](https://gist.github.com/di72nn/c54d83f3ebac0136983ea218b0ab7337) would probably fit best).

**Comment by zackw** [comment]
I don't think you understand the situation; for unrelated reasons the _only_ way I can bulk export a whole bunch of open tabs from the browser on my phone is via "share".  So, if multi-share to the wallabag Android client can't be implemented, I'm up a creek.  I don't see why the Android client _couldn't_ execute the API loop you describe.

**Comment by di72nn** [comment]
> for unrelated reasons the only way I can bulk export a whole bunch of open tabs from the browser on my phone is via "share".

Share to an email client (or chat application or text editor, etc.), send it to yourself, deal with it on a PC :man_shrugging: 

> So, if multi-share to the wallabag Android client can't be implemented, I'm up a creek.

I fail see it that way.


> I don't see why the Android client _couldn't_ execute the API loop you describe.

As far as I remember it can and does, the only missing thing is the share handling (not much code, but will probably need some UI to confirm multiple URLs so one does not accidentally clog their wallabag instance). Even if it were to be done now, it would take some time before a release, that's why I'm suggesting alternatives that you should be able to implement yourself without any delays.

**Comment by zackw** [comment]
Oh, I see.  Sorry for misunderstanding you.  I actually did manually export them all already so it's moot.
